### PR TITLE
Add path_prefix as env var

### PR DESCRIPTION
**Pull request by DatGuyJonathan**
On the hosting side: https://github.com/514-labs/commercial/pull/76

### CODE REVIEW

**Review by callicles — COMMENTED** [review]
Did you check that this was not breaking metrics as well?

File: apps/framework-cli/src/cli/local_webserver.rs [hunk 1]
```diff
@@ -805,11 +814,17 @@ impl Webserver {
         );
 
         if !project.is_production {
+            let path_prefix = project
+                .http_server_config
+                .path_prefix
+                .as_deref()
+                .map(|p| format!("/{}", p.trim_matches('/')))
+                .unwrap_or_default();
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/local_webserver.rs]: a bit weird that it's `.unwrap_or_default()` here `.unwrap_or("/".to_string())` several lines down
we can probably use the same variable.

File: apps/framework-cli/src/cli/local_webserver.rs [hunk 2]
```diff
@@ -627,12 +637,11 @@ async fn router(
...
 
-    let route_prefix = PathBuf::from("/");
-    let route = PathBuf::from(req.uri().path())
-        .strip_prefix(route_prefix)
-        .unwrap()
-        .to_path_buf()
-        .clone();
+    let original_path = PathBuf::from(req.uri().path());
+    let route = original_path
+        .strip_prefix(PathBuf::from(path_prefix))
```
Comment (phiSgr) [inline apps/framework-cli/src/cli/local_webserver.rs]: do we have to validate whether the prefix ends with `/`

File: apps/framework-cli/src/cli/local_webserver.rs (hunk 1 above)
Comment (DatGuyJonathan) [reply apps/framework-cli/src/cli/local_webserver.rs]: Updated, lmk if there's anything else.

File: apps/framework-cli/src/cli/local_webserver.rs (hunk 2 above)
Comment (DatGuyJonathan) [reply apps/framework-cli/src/cli/local_webserver.rs]: Updated, lmk if there's anything else.

**Comment by DatGuyJonathan** [comment]
https://github.com/user-attachments/assets/2a184243-9e85-42e3-aba2-4e7cd1fa4fc8

@callicles , metrics look good. As for sdk, I think we should be okay because the constructor takes a base url and I'd expect that to include the path prefix.

**Review by callicles — APPROVED** [review]

**Pull request merged**